### PR TITLE
Add alert for New Envelopes scheduled job

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-1021

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: scheduled-jobs-alert.tf [hunk 1]
```diff
@@ -111,3 +111,35 @@ module "delete-rejected-files-alert" {
...
+  app_insights_query =  <<EOF
+traces
+| where message startswith "Started check-new-envelopes job"
+| union (datatable(metric: real) [
+  0,
+]
+| extend timeNow = now()
+| extend day_of_week = toint(substring(tostring(dayofweek(timeNow)), 0, 1))
+| extend hour_of_the_day = datetime_part("hour", timeNow)
+| where day_of_week == 0 or day_of_week == 6 or hour_of_the_day < 9 or hour_of_the_day  > 17)
```
Comment (nitinprabhu) [inline scheduled-jobs-alert.tf]: what about your thoughts on below query ?

```
let startDatetime = startofday(now()); // 12AM
let startNotification = startDatetime + 9hours;
let stopNotification = startDatetime + 17hours;
let saturday = "6.00:00:00"; // Monday(0)
let sunday = "7.00:00:00";
traces
| where message startswith "Started check-new-envelopes job"
| where timestamp  >= startNotification and timestamp  <=stopNotification
| extend dayVar = dayofweek(timestamp)
| where dayVar != saturday and dayVar != sunday
| summarize count()

```

Comment (doncem) [reply scheduled-jobs-alert.tf]: no this query is actually not good:
- timestamp filter should be first citizen of the query
- we then check every hour for the interval between 9am 5pm. every hour (!)
- there's nothing to extend if there are no rows coming back from that query. hence another where will not work.

logically it'll always fail when it is weekend and before first envelope gets uploaded/processed during defined business hours

File: scheduled-jobs-alert.tf [hunk 2]
```diff
@@ -111,3 +111,38 @@ module "delete-rejected-files-alert" {
...
+| union (datatable(metric: real) [
+  0, // to mimic the non business hours
+]
+| extend timeNow = now()
+| extend day_of_week = toint(substring(tostring(dayofweek(timeNow)), 0, 1))
+| extend hour_of_the_day = datetime_part("hour", timeNow)
+| where day_of_week == 0 or // sunday
+        day_of_week == 6 or // saturday
+        hour_of_the_day < 9 or // scheduled job start hour
+        hour_of_the_day  > 17) // scheduled job end hour
```
Comment (rkondratowicz) [inline scheduled-jobs-alert.tf]: from acceptance criteria:
>If no envelopes are found then log with an **appropriate message**. (Check BSP)
>Define a new alert that would be triggered **if the message is present**

Can we simply send an email if a warning is logged by the job?

Comment (rkondratowicz) [reply scheduled-jobs-alert.tf]: ie:
```
traces
| where message startswith "No Envelopes created"
```
```

Comment (aliveni) [reply scheduled-jobs-alert.tf]: The alerts for this is already created in a separate PR 
https://github.com/hmcts/reform-scan-shared-infra/pull/34.
This PR is to make sure the scheduled job (https://github.com/hmcts/blob-router-service/pull/341) is running in business hours (the cron for the scheduled job is in https://github.com/hmcts/blob-router-service/pull/341)

Comment (rkondratowicz) [reply scheduled-jobs-alert.tf]: ok, I got confused by the PR title.

**Comment by aliveni** [comment]
Updated check-new-envelopes scheduled job cron to run every hour. https://github.com/hmcts/blob-router-service/pull/368

**Pull request merged**